src/models/Command.js

```diff
@@ -65,6 +65,7 @@ class Command {
             userJoinedChannel: self.onUserJoinedChannel || null,
 
             messageDelivered: self.onMessageDelivered || null,
+            preValidate: self.onPreValidate || null,
```

Comment:
Added a event that gets fired when we first recieve a message. before any validation.